src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -164,82 +195,66 @@ static int gavl_intersect_cmp_func(uintptr_t ustart, uintptr_t len, gavl_tree_no
 int MPL_gavl_tree_create(void (*free_fn) (void *), MPL_gavl_tree_t * gavl_tree)
 {
     int mpl_err = MPL_SUCCESS;
-    gavl_tree_s *gavl_tree_iptr;
+    gavl_tree_s *tree_ptr;
 
-    gavl_tree_iptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
-    if (gavl_tree_iptr == NULL) {
+    tree_ptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
+    if (tree_ptr == NULL) {
         mpl_err = MPL_ERR_SHM_NOMEM;
         goto fn_fail;
     }
 
-    gavl_tree_iptr->root = NULL;
-    gavl_tree_iptr->gavl_free_fn = free_fn;
-    *gavl_tree = (MPL_gavl_tree_t) gavl_tree_iptr;
+    tree_ptr->root = NULL;
+    tree_ptr->gavl_free_fn = free_fn;
+    TREE_CLEAR_STACK(tree_ptr);
```

Comment:
Since it will be called at beginning of every search, it is not necessary here, right? If you use `MPL_calloc` earlier, you can leave many fields implicitly zero. It is cleaner since zeroing the fields are not interesting to read.